### ISSUE TITLE
Test failure baseservices/threading/interlocked/exchange/exchangedouble/exchangedouble.sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#51849]
Run: [runtime-coreclr gcstress0x3-gcstress0xc 20210425.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1106740&view=ms.vss-test-web.build-test-results-tab&runId=33792530&resultId=103248&paneView=debug)

Failed test:
```
CoreCLR OSX arm64 Checked gcstress0xc @ OSX.1100.ARM64.Open
 -baseservices/threading/interlocked/exchange/exchangedouble/exchangedouble.sh
```

**Error message:**
```
Return code: 1
Raw output file: /private/tmp/helix/working/A9C9098C/w/BD760A43/e/baseservices/threading/Reports/baseservices.threading/interlocked/exchange/exchangedouble/exchangedouble.output.txt
Raw output:
BEGIN EXECUTION
/tmp/helix/working/A9C9098C/p/corerun exchangedouble.dll ''
-9.255963134931783E+61,-1.7976931348623157E+308,0
Test Failed
Expected: 100
Actual: 0
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/tmp/helix/working/A9C9098C/p
> /private/tmp/helix/working/A9C9098C/w/BD760A43/e/baseservices/threading/interlocked/exchange/exchangedouble/exchangedouble.sh
Expected: True
Actual: False

Stack trace
   at baseservices_threading._interlocked_exchange_exchangedouble_exchangedouble_._interlocked_exchange_exchangedouble_exchangedouble_sh()

```

**Comment by BruceForstall** [comment]
@sdmaclea

**Comment by mangod9** [comment]
believe these should be fixed after upgrading to 11.3
